From d75c955357c9eea98727d082b883dcb9977e1f29 Mon Sep 17 00:00:00 2001
Message-ID: <d75c955357c9eea98727d082b883dcb9977e1f29.1719691458.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1719691458.git.sharlatanus@gmail.com>
References: <cover.1719691458.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 24 Jun 2024 21:48:32 +0100
Subject: [PATCH 24/26] gnu: Add go-github-com-pterm-pterm.

* gnu/packages/golang-xyz.scm (go-github-com-pterm-pterm): New variable.

Change-Id: I6da814b83833224380a5dd08da9cf1562eb33791
---
 gnu/packages/golang-xyz.scm | 38 +++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 7033618295..80d68f2fd3 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -3950,6 +3950,44 @@ (define-public go-github-com-prometheus-client-model
     (home-page "https://github.com/prometheus/client_model")
     (license license:asl2.0)))
 
+(define-public go-github-com-pterm-pterm
+  (package
+    (name "go-github-com-pterm-pterm")
+    (version "0.12.79")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/pterm/pterm")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0xvc4ywc2998r8vsi3zpp49z04kc79q60bsvxv88cjvamxfjxrvk"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      ;; Cycle: go-github-com-pterm-pterm -> go-github-com-marvinjwendt-testza
+      ;; -> go-github-com-pterm-pterm
+      #:tests? #f
+      #:go go-1.21
+      #:import-path "github.com/pterm/pterm"))
+    (propagated-inputs
+     (list go-atomicgo-dev-cursor
+           go-atomicgo-dev-keyboard
+           go-atomicgo-dev-schedule
+           go-github-com-gookit-color
+           go-github-com-lithammer-fuzzysearch
+           go-github-com-mattn-go-runewidth
+           go-golang-org-x-term
+           go-golang-org-x-text))
+    (home-page "https://github.com/pterm/pterm")
+    (synopsis "Configurable consol outputs in Golang")
+    (description
+     "Package pterm is a modern go module to beautify console output.  It can be used
+without configuration, but if desired, everything can be customized down to the
+smallest detail.")
+    (license license:expat)))
+
 (define-public go-github-com-rcrowley-go-metrics
   (let ((commit "cac0b30c2563378d434b5af411844adff8e32960")
         (revision "2"))
-- 
2.41.0

